client/client.go

```diff
@@ -106,6 +107,10 @@ func New(
 		entries:   make(map[uint]watcherEntry),
 		condition: c.matchFundingProposal,
 	}
+	c.settlementWatcher = &stateWatcher{
+		entries:   make(map[uint]watcherEntry),
+		condition: c.matchSettlementProposal,
+	}
```

Comment:
Please add `newStateWatcher(condition) *stateWatcher` constructor to avoid repetition of internals (making a map).